### ISSUE TITLE
Nullable reference types: No unreachable warning for null check on not-nullable variable

### ISSUE THREAD

**Issue by bkoelman** [dotnet/roslyn#9980]
I have been playing with nullable reference types, which is already working remarkably great! To try it out, I have [patched up](https://github.com/bkoelman/ResharperCodeContractNullability/tree/staticnull) my analyzer to convert Resharper's `(Item)NotNull/CanBeNullAttribute` annotations to nullable reference types. Then ran that on an [actual annotated codebase](https://github.com/bkoelman/DogAgilityCompetitionManagement). 

During that process I discovered some oddities, which I'll create issues for here.

**Version Used**: Built from source, branch NullableReferenceTypes (commit https://github.com/dotnet/roslyn/commit/8553cdac75c2b0901e8ae401aaa23022f0689935)

**Steps to Reproduce**:
The next block of code does not raise any warnings:

``` csharp
void Test()
{
    const string s = "x";

    if (s == null)
    {
        throw new Exception();
    }
}
```

**Expected Behavior**: A warning that indicates the null check will never evaluate to `true`; possibly reuse [CS0472](https://msdn.microsoft.com/en-us/library/ms228621%28v=vs.90%29.aspx). Followed by [CS0162](https://msdn.microsoft.com/en-us/library/c0h4st1x.aspx), which indicates the `throw` line is unreachable.

**Actual Behavior**:
No warnings.

@AlekseyTs

**Comment by AlekseyTs** [comment]
@bkoelman Is the actual behavior accurate? There are no warnings reported at all?

**Comment by AlekseyTs** [comment]
And no errors?

**Comment by bkoelman** [comment]
This is what I get (running in VS2015 Update1):

![image](https://cloud.githubusercontent.com/assets/10324372/13969975/35852396-f086-11e5-91dd-d97df9f012d1.png)

**Comment by bkoelman** [comment]
Seeing it back, the picture is hard to read. The warning line contains `5`, not `s`.

**Comment by AlekseyTs** [comment]
@MadsTorgersen If I remember correctly, we decided against reporting warnings for null checks against values we "know" are not null. The lack of CS0162 is not related to "Nullable Reference Types" feature.

**Comment by bkoelman** [comment]
This repros in [Nullable Reference Types Preview 3/6/18](https://github.com/dotnet/csharplang/wiki/Nullable-Reference-Types-Preview).

I'm surprised about that decision, because the description for [CS0472](https://docs.microsoft.com/en-us/previous-versions/visualstudio/visual-studio-2008/ms228621(v=vs.90)) literally contains:

*The compiler should warn if you use an operator with a constant null value.*

Can I ask what the motivation was against emitting a warning in this case?

**Comment by gafter** [comment]
Issue moved to [dotnet/csharplang #1953](https://github.com/dotnet/csharplang/issues/1953) via [**ZenHub**](https://www.zenhub.com/)
